CRM/Queue/Queue/SqlTrait.php

```diff
@@ -55,12 +55,12 @@ public function existsQueue() {
    *
    * @return int
    */
-  public function numberOfItems() {
-    return CRM_Core_DAO::singleValueQuery("
+  public function numberOfItems(): int {
```

Comment:
I suppose this is safe from the POV that concrete function `numberOfItems(): int` still satisfies `CRM_Queue_Queue`'s abstract `numberOfItems()`. 

However, there are downstream classes ([civirules](https://lab.civicrm.org/extensions/civirules/-/blob/2.43/CRM/Queue/Queue/Civirules.php#L3-10), [form-processor](https://lab.civicrm.org/extensions/form-processor/-/blob/1.39/CRM/Queue/Queue/FormProcessor.php#L3-10)) which extend/override `numberOfItems()`, and this raises an error:

```
Fatal error: Declaration of CRM_Queue_Queue_FormProcessor::numberOfItems() must be
compatible with CRM_Queue_Queue_Sql::numberOfItems(): int 
in /...SNIP.../ext/form-processor/CRM/Queue/Queue/FormProcessor.php on line 10
```

Is it functionally important to change the return-type declaration?

Reply:
no - not functionally important - but I think that is going to be a thing in time with the way php is going - we can PR them to return an int since we've done the work of figuring them out

Reply:
How do they get away with not defining visibility :-(

Reply:
I'm pretty sure default visibility is `public`. (IIRC, PHP `class`es originally didn't allow visibility flags; the original behavior was semantically-equivalent to `public`. So `public` is a natural default.)

> we can PR them to return an int since we've done the work of figuring them out

OK, I think the reverse is safe -- in the class-tree, descendents can declare tighter return-types than ancestors. So if we can get descendent classes to roll out `numberOfItems(): int` before we freeze, then that would give time for users to start updating the ext first.

OTOH, if we have to do that kind of coordination, it would be tempting to change/fix other parts of the contract at the sam etime... like... `numberOfItems()` doesn't distinguish between "ripe/near-term items" and "deferred/future items"... which appears to be why it's overridden...

Reply:
Hmm - in that case we should add a new function & deprecate the existing I guess

Reply:
@totten - I've removed this type hint - I think we could add in the sql `   and (release_time is null OR release_time <= NOW())` here & then the override could go - however it would be good to get this merged cost imports are superficially broken in master & we have someone trying to test on there & having team members trying to test on there to find a member import bug & the troubleshooting went down a rabbit hole yesterday due to this